lib/matplotlib/patheffects.py

```diff
@@ -373,3 +375,167 @@ def draw_path(self, renderer, gc, tpath, affine, rgbFace):
         if clip_path:
             self.patch.set_clip_path(*clip_path)
         self.patch.draw(renderer)
+
+
+class TickedStroke(AbstractPathEffect):
+    """
+    A line based PathEffect which draws a path with a ticked style.
+    This line style is frequently used to represent constraints in
+    optimization.  The ticks indicate that one side of the line is
+    invalid.
+
+    The spacing, length, and angle of ticks can be controlled.
+
+    This line style is sometimes referred to as a hatched line, but
+    in matplotlib, hatching typically refers to filling an area with
+    a line pattern.
+
+    """
+
+    def __init__(self, offset=(0, 0),
+                 spacing=10.0, angle=45.0, length=np.sqrt(2),
+                 flip=False, **kwargs):
+        """
+        Parameters
+        ----------
+        offset : pair of floats
+            The offset to apply to the path, in points.
+        spacing : float
+            The spacing between ticks in points.
+            Default is 10.0.
+        angle : float
+            The angle between the path and the tick in degrees.
```

Comment:
Can you specify that this angle is relative to the figure rather than data?  I guess you also need to explain how angle is defined?  Is it the same if a line goes from right to left or left to right?

Reply:
Does a46275e below take care of it?  To me, the important part is the convention with respect to contour lines.  I'm relying on [this comment](https://github.com/matplotlib/matplotlib/blob/master/src/_contour.h#L66) to set the convention, but I'm not sure how to link there from the documentation -- or if that is overkill.

Reply:
And of course, the Flip option switches it up, but that is mentioned a few lines down.  I wasn't sure about mentioning it in the angle description.